test/src/unit.cc

```diff
@@ -61,7 +99,13 @@ int main(const int argc, char** const argv) {
   if (rc != 0)
     return rc;
 
-  return session.run();
+  int return_code = session.run();
+
+  // Run logger validation tests
```

Comment:
The motivation of adding it to the main function is that the logs should be tested after all the tests of a suite is done, and I think running the logger tests on their own doesn't really make sense. In an abstraction sense, I do understand that it would be nice to just run the logger tests on their own, but overall the logger tests should have input and that's best done running them with another suite of unit tests. However, I think it would be nice to have a flag that states whether a user wants to run logging tests. That would also solve the oss_sink problem too.

Reply:
> I think running the logger tests on their own doesn't really make sense

If you're considering a test of the logger as a unit test, this is exactly what you want to do, which is to run tests on the logger outside of all the tests.

If you're considering a test of the logger as an integration test, then you want what you've done here. This PR is missing documentation that this is an integration test, not a unit test.